src/controllers/controller.bar.js

```diff
@@ -496,16 +498,20 @@ export default class BarController extends DatasetController {
 
 }
 
-BarController.prototype.dataElementType = Rectangle;
-
-BarController.prototype.dataElementOptions = [
-	'backgroundColor',
-	'borderColor',
-	'borderSkipped',
-	'borderWidth',
-	'barPercentage',
-	'barThickness',
-	'categoryPercentage',
-	'maxBarThickness',
-	'minBarLength'
-];
+BarController.id = 'bar';
+BarController.defaults = defaults;
+BarController.preRegister = () => {
```

Comment:
Could use `beforeRegister` and `afterRegister`? That way we'd match the format of other lifecycle calls

Reply:
I agree tho those names and we could have those hooks, but I think we should not expect anything to happen in beforeRegister/afterRegister. Maybe we can come up with a better name for this method. `initializePrototype` (just to get you thinking, I don't like that either)